closed/autoconf/custom-hook.m4

```diff
@@ -790,3 +791,70 @@ AC_DEFUN([OPENJ9_GENERATE_TOOL_WRAPPERS],
   OPENJ9_GENERATE_TOOL_WRAPPER([nasm], [$NASM])
   OPENJ9_GENERATE_TOOL_WRAPPER([rc], [$RC])
 ])
+
+AC_DEFUN([OPENJCEPLUS_SETUP],
+[
+  BUILD_JGSKIT=false
+  GSKIT_BIN=
+  GSKIT_SDK_BIN=
+  GSKIT_BIN_CREDENTIAL=
+  GSKIT_HOME=
+  OPENJCEPLUS_TOPDIR=
+
+  # check for GSKIT crypyo library for OpenJCEPlus
+  AC_ARG_WITH(gskit-bin, [AS_HELP_STRING([--with-gskit-bin], [specify path to the gskit bin url])])
+  AC_ARG_WITH(gskit-sdk-bin, [AS_HELP_STRING([--with-gskit-sdk-bin], [specify path to the gskit sdk bin url])])
+  AC_ARG_WITH(gskit-bin-credential, [AS_HELP_STRING([--with-gskit-bin-credential], [specify path to the gskit bin credential])])
+
+  AC_MSG_CHECKING([if gskit and gskit sdk bin url is set])
+  if test "x$with_gskit_bin" = xno || test "x$with_gskit_sdk_bin" = xno ; then
+    AC_MSG_RESULT([no])
+  elif test "x$with_gskit_bin" = x || test "x$with_gskit_bin" = xyes || test "x$with_gskit_sdk_bin" = x || test "x$with_gskit_sdk_bin" = xyes ; then
+    AC_MSG_RESULT([no])
+    PRINT_GSKIT_HELP
+    AC_MSG_ERROR([--with-gskit-bin and --with-gskit-sdk-bin must have values])
+  else
+    AC_MSG_RESULT([yes])
+    BUILD_JGSKIT=true
+    GSKIT_BIN="$with_gskit_bin"
+    GSKIT_SDK_BIN="$with_gskit_sdk_bin"
+
+    AC_MSG_CHECKING([if gskit and gskit sdk bin credential is set])
+    if test "x$with_gskit_bin_credential" != x ; then
+      AC_MSG_RESULT([yes])
+      GSKIT_BIN_CREDENTIAL="$with_gskit_bin_credential"
+    else
+      AC_MSG_RESULT([no])
+    fi
+
+    OPENJCEPLUS_TOPDIR="$TOPDIR/OpenJCEPlus"
+    GSKIT_HOME="$TOPDIR/OpenJCEPlus/icc/jgsk_sdk"
+    if ! test -d "$OPENJCEPLUS_TOPDIR" ; then
+      AC_MSG_ERROR(["Cannot locate the path to OpenJCEPlus sources: $OPENJCEPLUS_TOPDIR! Try 'bash get_source.sh' and restart configure"])
+    fi
+  fi
+
+  AC_SUBST(BUILD_JGSKIT)
+  AC_SUBST(GSKIT_BIN)
+  AC_SUBST(GSKIT_SDK_BIN)
+  AC_SUBST(GSKIT_BIN_CREDENTIAL)
+  AC_SUBST(GSKIT_HOME)
+  AC_SUBST(OPENJCEPLUS_TOPDIR)
+])
+
+AC_DEFUN([PRINT_GSKIT_HELP],
+[
+  printf "\n"
+  printf "GSKIT is required to compile the OpenJCEPlus native library and has to\n"
+  printf "be provided during configure process in order to build OpenJCEPlus with\n"
+  printf "GSKIT.\n"
+  printf "\n"
+  printf "Run configure with '--with-gskit-bin' and '--with-gskit-sdk-bin' to\n"
+  printf "specify the URL of GSKIT binary and GSKIT SDK binary. And run configure\n"
+  printf "with '--with-gskit-bin-credential' to specify the credential(if needed)\n"
+  printf "when downloading the GSKIT binary and GSKIT SDK binary.\n"
+  printf "\n"
+  printf "To ignore OpenJCEPLUs with GSKIT, run configure with '--with_gskit_bin=no',\n"
+  printf "or '--with-gskit-sdk-bin=no'."
```

Comment:
The default should be to ignore OpenJCEPlus (typo in `OpenJCEPLUs`); those options should only be required to choose some non-default configuration.